tests/pact/fqns.repl

```diff
@@ -77,3 +77,98 @@
 
 (expect "selects correct test" (modB.get-test) "hello")
 (commit-tx)
+
+;;
+;; Module redeploy name resolution
+;;
+
+
+(begin-tx)
+(env-exec-config ["DisablePact48"])
+
+(namespace 'free)
+(module test-mod-redeploy-ref g
+  (defcap g () true)
+
+  (defcap test ()
+    (enforce false "boom"))
+
+  (defun f ()
+    (with-capability (test)
+      1))
+  )
+(expect-failure "Execution fails due to the false defcap" (f))
```

Comment:
It's better if failure docs don't just say "it fails because xxx" but instead _why_ the case is being exercised. Here the comment `"Execution fails due to the false defcap"` copypasta'd to the three assertions don't explain the following:
- why is failure important? e.g. why not just show a cap succeeding? An example doc would be "Verify failure to show XXX"
- why are there three cases? the comments should help disambiguate the cases. Why do we think failure would not occur in one case but not another?